File: AOJ/0430/0430.c
#include<stdio.h>

int min(int a, int b){ return a < b ? a : b; }
int max(int a, int b){ return a > b ? a : b; }

int main(){
  int N, x, d, R, L, i, minv, maxv;
  scanf("%d %d", &N, &x);
  R = L = x;
  for ( i = 0; i < N; i++ ){
    scanf("%d", &d);
    R = max(R, d);
    L = min(L, d);
  }
  minv = min(R - x, x - L);
  maxv = max(R - x, x - L);
  printf("%d\n", minv * 2 + maxv);
  return 0;
}

File: AOJ/0430/0430.py
N = int(input())
X = list(map(int, input().split()))
x = X[0]
R = x
L = x
for i in range(1, N):
    R = max(R, X[i])
    L = min(L, X[i])
minv = min(R - x, x - L)
maxv = max(R - x, x - L)
print(minv * 2 + maxv)
